Add page types to Bandcamp

dillonupgradeit opened the pull request:
Additional page types include: singles, featured pages, tag/genre pages, fan/collection pages, homepage discover page.

dillonupgradeit committed:
Add Page Types to Bandcap

Additional page types include: singles, fan/collelction pages, featured pages, bandcamp weekly pages, and homepage discover.

dillonupgradeit committed:
Refactor Version Number

Refactor version number.

dillonupgradeit committed:
Refactor Cleanup

Refactor comments and spacing.

dillonupgradeit committed:
Merge from Master

Merge from master and resolve conflicts.

cmgriffing commented on get-song-name.js:
```diff
@@ -1,56 +1,161 @@
+
+// Bandcamp hompage discover section
+// https://bandcamp.com/
+var discover = document.querySelector('.discover-item .playing');
```
I think as a pattern we want to stay away from var and use let or const instead. var would be for strictly needing hoisting.

dillonupgradeit replied:
Resolved in commit 32dc34d

cmgriffing commented on get-song-name.js:
```diff
@@ -1,56 +1,161 @@
+  const albumName = discoverArray.querySelector('.item-title').textContent;
+  const albumURL = discoverArray.querySelector('.item-title').href;
+  const artist = discoverArray.querySelector('.item-artist').textContent;
+  const artistURL = discoverArray.querySelector('.item-title').href;
```
Should the selector on this line be pointing at the .item-artist href instead?

dillonupgradeit replied:
Resolved in commit a5364d7

dillonupgradeit committed:
Remove use of var from get-song-info.js

Remove use of var from get-song-info.js

dillonupgradeit committed:
Wrap get-song-info.js in function

Wrap get-song-info.js in function to avoid redeclared constants error.

dillonupgradeit committed:
Update albumTitle node on single/track pages

Update albumTitle node on single/track pages with backup itemprop nodes.

dillonupgradeit committed:
Update bandcamp weekly to include playlist name and url

dillonupgradeit committed:
Update get-song-info.js discover to utilize discover detail player

dillonupgradeit committed:
Fix issue with homepage and bcweekly-info-inner panel

Bandcamp hompage also displayed bcweekly-info-inner, so you cannot rely on this to determine if Bandcamp Weekly Show page.

dillonupgradeit committed:
Add artistURL to tag, fan, and collection pages

dillonupgradeit committed:
Fix issue with album URL on hompage discover

dillonupgradeit committed:
Add currentTime and durationTime

Add currentTime and durationTime to album + single/track pages, tags + fans + collections pages, and bandcamp featured page

dillonupgradeit committed:
Add commented descriptions to get-song-name.js

dillonupgradeit committed:
Fix currentTime and durationTime on Bandcamp weekly page

dillonupgradeit committed:
Update Version Number

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Update README.md

Update README.md

Update README.md

Update README.md

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Correct Version Number

dillonupgradeit committed:
Correct Version Number

dillonupgradeit committed:
Add Changelog to README.md

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Update get-song-info.js filename to clarify

dillonupgradeit committed:
Update get-song-info.js filename to clarify

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Merge branch 'Add-Page-Types-to-Bandcamp' of https://github.com/dillonupgradeit/now-playing into Add-Page-Types-to-Bandcamp
